src/core/dev_api/openvino/runtime/shared_buffer.hpp

```diff
@@ -28,4 +28,51 @@ class SharedBuffer : public ov::AlignedBuffer {
     T _shared_object;
 };
 
+/// \brief SharedStreamBuffer class to store pointer to pre-acclocated buffer and provide streambuf interface.
+class SharedStreamBuffer : public std::streambuf {
+public:
+    SharedStreamBuffer(char* data, size_t size) : m_data(data), m_size(size), m_offset(0) {}
+
+protected:
+    std::streamsize xsgetn(char* s, std::streamsize count) override {
+        auto real_count = std::min<std::streamsize>(m_size - m_offset, count);
+        std::memcpy(s, m_data + m_offset, real_count);
+        m_offset += real_count;
+        return real_count;
+    }
+
+    int_type underflow() override {
+        return (m_size == m_offset) ? traits_type::eof() : traits_type::to_int_type(*(m_data + m_offset));
+    }
+
+    int_type uflow() override {
+        return (m_size == m_offset) ? traits_type::eof() : traits_type::to_int_type(*(m_data + m_offset++));
+    }
+
+    std::streamsize showmanyc() override {
+        return m_size - m_offset;
+    }
+
+    char* m_data;
+    size_t m_size;
+    size_t m_offset;
+};
+
+/// \brief OwningSharedStreamBuffer is a SharedStreamBuffer which owns its shared object. Can return AlignedBuffer to
+/// shared memory
+class OwningSharedStreamBuffer : public SharedStreamBuffer {
+public:
+    template <typename T>
+    OwningSharedStreamBuffer(char* data, size_t size, const T& shared_object)
+        : SharedStreamBuffer(data, size),
+          m_alligned_buffer(std::make_shared<SharedBuffer<T>>(data, size, shared_object)) {}
+
+    std::shared_ptr<AlignedBuffer> get_aligned_buffer() {
```

Comment:
Why is required and why not just return internal buffer. When shared buffer used it will just increment counter in stored shared object.

Reply:
Just to reuse owning logic from SharedBuffer

Reply:
But you're right. There is no need to hold AlignedBuffer. std::shared_ptr<void> is enough